### PR TITLE
Chore: Use the promlib v0.0.1

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

This is a complementary PR that enables us to use versioned `promlib`

Prometheus Library https://github.com/grafana/grafana/pull/83952

### CODE REVIEW

**Review by toddtreece — APPROVED** [review]

File: pkg/promlib/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# promlib
+
+Prometheus Library a.k.a. (promlib) is the foundation of the Grafana prometheus data source backend.
+
+### How to tag/version?
+- Checkout the commit you want to tag (git checkout <COMMIT_SHA>)
```
Comment (NWRichmond) [inline pkg/promlib/README.md:6]: nit: per `markdownlint` [md001](https://github.com/DavidAnson/markdownlint/blob/v0.33.0/doc/md001.md) and [md022](https://github.com/DavidAnson/markdownlint/blob/v0.33.0/doc/md022.md) / [md032](https://github.com/DavidAnson/markdownlint/blob/v0.33.0/doc/md032.md),

```suggestion
## How to tag/version?

- Checkout the commit you want to tag (git checkout <COMMIT_SHA>)
```

File: pkg/promlib/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# promlib
+
+Prometheus Library a.k.a. (promlib) is the foundation of the Grafana prometheus data source backend.
```
Comment (NWRichmond) [inline pkg/promlib/README.md:3]: nit:

```suggestion
Prometheus Library (a.k.a. promlib) is the foundation of the Grafana Prometheus data source backend.
```

File: pkg/promlib/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+### How to tag/version?
+- Checkout the commit you want to tag (git checkout <COMMIT_SHA>)
+- Run git tag <VERSION> (For example v0.123.0)
```
Comment (NWRichmond) [inline pkg/promlib/README.md:7]:
```suggestion
- Run `git tag <VERSION>` (For example v0.123.0)
```

File: pkg/promlib/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  - NOTE: We're using Lightweight Tags, so no other options are required
+- Run git push origin <VERSION>
```
Comment (NWRichmond) [inline pkg/promlib/README.md:9]:
```suggestion
- Run `git push origin <VERSION>`
```

**Review by NWRichmond — APPROVED** [review]
Looks good! Just a few nitpicky suggestions.

File: go.mod [hunk 5]
```diff
@@ -513,3 +514,5 @@ exclude github.com/mattn/go-sqlite3 v2.0.3+incompatible
 
 // Use our fork xorm. go.work currently overrides this and points to the local ./pkg/util/xorm directory.
 replace xorm.io/xorm => github.com/grafana/grafana/pkg/util/xorm v0.0.1
+
+replace github.com/grafana/grafana/pkg/promlib => ./pkg/promlib
```
Comment (andresmgot) [inline go.mod:518]: I'd add a comment here explaining this replace (as with the rest of replaces)

**Review by andresmgot — APPROVED** [review]
LGTM

**Review by toddtreece — APPROVED** [review]

**Review by kylebrandt — APPROVED** [review]

**Pull request merged**